File: 01-cajero/main.py
disponible = {
    100: 3,
    50: 6,
    20: 10,
    10: 50,
    1: 50,
}

def total_disponible():
    total = 0
    for billete, cantidad in disponible.items():
   		total += billete * cantidad
    return total


def retirar(monto):
    if monto > total_disponible():
        print("Error, no hay suficiente efectivo")
    
    saldo = monto
    while saldo > 0:
        for denominacion, v in disponible.items():
            cantidad = get_denominacion(denominacion, saldo)
            print("Billete o moneda de: ", denominacion, cantidad)
            saldo -= cantidad * denominacion
        

def get_denominacion(denominacion, saldo):
    cantidad = saldo/denominacion
    if (cantidad > disponible[denominacion]):
        cantidad = disponible[denominacion]
    
    disponible[denominacion] -= cantidad
    return cantidad


retirar(1300)

File: 01-cajero/main.js
const disponible = {
    100: 3,
    50: 6,
    20: 10,
    10: 50,
    1: 50,
};

const billetes = Object.keys(disponible);

const totalDisponible = () => {
    let total = 0;
    for (let i = 0; i < billetes.length; i++) {
        total += billetes[i] * disponible[billetes[i]]
    }

    return total;
}


const calcularBilletes = (saldoARetirar, denominacion) => {
    let contador = saldoARetirar / denominacion

    if (contador > disponible[denominacion]) {
        contador = disponible[denominacion]
    }
    disponible[denominacion] -= contador

    return contador
}


const procesar = valorAretirar => {
    if (valorAretirar > totalDisponible()) {
        console.log("Error, no hay suficiente saldo")

        return
    }

    let saldoARetirar = valorAretirar;
    while (saldoARetirar > 0) {

        let contador = 0;
        for (let i = 0; i < billetes.length; i++) {
            contador = calcularBilletes(saldoARetirar, billetes[i]);
            console.log(`${contador} billetes de denonimacion: ${billetes[i]}`);

            saldoARetirar -= contador * billetes[i]
        }
    }
}



procesar(135);
